contrib/automation_tests/test_cases/capture_window.py

```diff
@@ -188,9 +188,22 @@ def _show_capture_window(self):
         logging.info('Showing capture window')
         self.find_control("TabItem", "Capture").click_input()
 
-    def _execute(self, length_in_seconds: int = 5):
-        self._show_capture_window()
+    def _set_capture_options(self, collect_thread_states: bool):
+        capture_tab = self.find_control('Group', "CaptureTab")
+
+        logging.info('Opening "Capture Options" dialog')
+        capture_options_button = self.find_control('Button', 'Capture Options', parent=capture_tab)
+        capture_options_button.click_input()
+
+        collect_thread_states_checkbox = self.find_control('CheckBox', 'Collect thread states')
+        if collect_thread_states_checkbox.get_toggle_state() != collect_thread_states:
+            logging.info('Toggling "Collect thread states" checkbox')
+            collect_thread_states_checkbox.click_input()
+
+        logging.info('Saving "Capture Options"')
+        self.find_control('Button', 'OK').click_input()
```

Comment:
If, as commented below, the dialog is really parented underneath the top window, 'OK' is a likely candidate to be ambiguous. Would be more robust to pass the dialog as explicit parent to all `find_control` calls in this method.

Reply:
I think you are right, I'll try to get the "Capture Options" window and pass it to `find_control`.

Reply:
Done.